### PR TITLE
WIP: internal/test helpers

### PR DESCRIPTION
**Pull request by imjasonh**
Opening this to solicit feedback.

It's internal now, after we settle on an API we all like I'll take a pass at applying it to our own tests, and eventually make this an external API so folks can use it in their own tests (myself included 😆 )

### CODE REVIEW

File: internal/test/must.go [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+func T(t *testing.T) Muster {
+	return testMuster{t}
+}
+
+type digester interface {
+	Digest() (v1.Hash, error)
+}
+
+type Muster interface {
+	RandomImage(int64, int64) v1.Image
```
Comment (jonjohnsonjr) [inline internal/test/must.go:49]: I'd rather not take a dep on `random` if we can avoid it.

Can this be `Image(v1.Image, error) v1.Image` called as `must.Image(random.Image(...))`?

Comment (imjasonh) [reply internal/test/must.go]: Great idea. Done!

File: internal/test/must.go [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+// test.T(t).Digest(img)
+//
+// or
+//
+// must := must.T(t)
+// must.Digest(img)
+
+// func TestFoo(t *testing.T) {
+//   must := test.T(t)
+//   img := must.RandomImage(1024, 5)
```
Comment (jonjohnsonjr) [inline internal/test/must.go:33]:
```suggestion
//   img := must.Image(random.Image(1024, 5))
```

Comment (imjasonh) [reply internal/test/must.go]: Dropped this comment section anyway, in favor of the test, which will actually stay up to date.

**Pull request closed without merging**